src/charm.py

```diff
@@ -255,38 +276,57 @@ def _ue_pebble_layer(self) -> Layer:
                     self._service_name: {
                         "override": "replace",
                         "startup": "enabled",
-                        "command": self._ue_startup_command,
+                        "command": self._get_ue_startup_command(rfsim=rfsim),
                         "environment": self._ue_environment_variables,
                     },
                 },
             }
         )
 
-    @property
-    def _ue_startup_command(self) -> str:
+    def _get_ue_startup_command(self, rfsim: bool) -> str:
+        if rfsim:
+            return " ".join(
+                [
+                    "/opt/oai-gnb/bin/nr-uesoftmodem",
+                    "-O",
+                    f"{BASE_CONFIG_PATH}/{CONFIG_FILE_NAME}",
+                    "--sa",
+                    "--rfsim",
+                    "-r",
+                    "106",
+                    "--numerology",
+                    "1",
+                    "-C",
+                    "3924060000",
+                    "--ssb",
+                    "530",
+                    "--band",
+                    "77",
+                    "--log_config.global_log_options",
+                    "level,nocolor,time",
+                    "--rfsimulator.serveraddr",
+                    str(self.rfsim_requirer.rfsim_address)
+                    if self.rfsim_requirer.rfsim_address
+                    else "",
+                ]
+            )
         return " ".join(
             [
                 "/opt/oai-gnb/bin/nr-uesoftmodem",
                 "-O",
                 f"{BASE_CONFIG_PATH}/{CONFIG_FILE_NAME}",
                 "--sa",
-                "--rfsim",
                 "-r",
                 "106",
                 "--numerology",
                 "1",
-                "-C",
-                "3924060000",
                 "--ssb",
                 "530",
                 "--band",
                 "77",
-                "--log_config.global_log_options",
-                "level,nocolor,time",
```

Comment:
Why we don't want these when using a real radio?

Reply:
No good reason, I was using the value in the docs, I added those in the last commit